js/src/volume.js

```diff
@@ -485,32 +502,103 @@ var ScatterView = widgets.WidgetView.extend( {
         // colors
         var colors = new THREE.InstancedBufferAttribute(new Float32Array( max_count * 3 ), 3, 1);
         var colors_previous = new THREE.InstancedBufferAttribute(new Float32Array( max_count * 3 ), 3, 1);
-
         function to_rgb(color) {
             color = new THREE.Color(color)
             return [color.r, color.g, color.b]
         }
-        var color = to_rgb(this.model.get("color"))
-        var color_previous = "color" in this.previous_values ? to_rgb(this.previous_values["color"]) : color;
+
+        function get_value_index_color(variable,index,max_count){
+            //Return a two D array (max_count,3)
+            //First find type and Dimension
+
+            var vartype = 0;  //1 for string
+            if ( typeof variable == "string" || typeof variable[0] == "string" || typeof variable[0][0] == "string") {
+                vartype = 1;
+            }
+
+            if (vartype == 1) {
+                if ( typeof variable == "string") {
+                //OD
+                    color =  to_rgb(variable)
+                    return color
+                }
```

Comment:
I suggest refactoring this a bit, so it's easier to read/maintain.
instead of returning the color, and checking the color dim, you could do sth like this:
```javascript
color = to_rgb(variable)
return function(glyph_index) {
   return color
}